test/parallel_api/algorithm/alg.modifying.operations/transform_binary_zip.pass.cpp

```diff
@@ -0,0 +1,152 @@
+// -*- C++ -*-
+//===-- transform_binary.pass.cpp -----------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include _PSTL_TEST_HEADER(execution)
+#include _PSTL_TEST_HEADER(algorithm)
+
+#include "support/utils.h"
+
+using namespace TestUtils;
+
+template <typename... Types>
+using typle_t = oneapi::dpl::__internal::tuple<Types...>;
+//using typle_t = ::std::tuple<Types...>;
```

Comment:
@rarutyun May you take a look why I unable to use `td::tuple` in this test?
I have compile error in this case.

Reply:
In other tests I found the code like in https://github.com/oneapi-src/oneDPL/blob/5fe797ff97db04e7446448d0177ce376380d7b59/test/parallel_api/iterator/iterators.pass.cpp#L177
But that's another algorithm and unclear what is the error in our tests code.

Reply:
Problem fixed together with @kboyarinov in commit https://github.com/oneapi-src/oneDPL/pull/1274/commits/f2b67b228b6f8ef2bb2bf13454f6925a60b15d6b

The root case was in implementation the the `operator()` :
```C++
    template <typename T1, typename T2>
    U operator()(const typle_t<T1, T2>& x) const
    {
        return U(val - std::get<0>(x) - std::get<1>(x));
    }
```
- in this code we should have concrete type of `tuple` arguments.